src/sentry/api/endpoints/organization_sampling_project_rates.py

```diff
@@ -0,0 +1,92 @@
+from collections.abc import Mapping, MutableMapping
+from typing import Any
+
+from django.db import router, transaction
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint, OrganizationPermission
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import Serializer, serialize
+from sentry.constants import TARGET_SAMPLE_RATE_DEFAULT, ObjectStatus
+from sentry.models.options.project_option import ProjectOption
+from sentry.models.project import Project
+
+OPTION_KEY = "sentry:target_sample_rate"
+
+
+class GetSerializer(Serializer):
+    """Serializer for OrganizationSamplingProjectRatesEndpoint.get"""
+
+    def get_attrs(self, item_list, user, **kwargs) -> MutableMapping[Any, Any]:
+        options = ProjectOption.objects.get_value_bulk(item_list, OPTION_KEY)
+        # NOTE: `get_value_bulk` does not resolve defaults. The default does not
+        # depend on epochs, so we can speed this up by using the constant.
+        return {
+            item: value if value is not None else TARGET_SAMPLE_RATE_DEFAULT
+            for item, value in options.items()
+        }
+
+    def serialize(self, obj: Any, attrs: Any, user, **kwargs) -> Mapping[str, Any]:
+        return {"id": obj.id, "sampleRate": attrs}
+
+
+class PutSerializer(serializers.Serializer):
+    """Serializer for OrganizationSamplingProjectRatesEndpoint.put"""
+
+    id = serializers.IntegerField(required=True)
+    sampleRate = serializers.FloatField(required=True, min_value=0, max_value=1)
```

Comment:
Why not have just one serializer, and do the logic from `get_attrs` inside the get method?

```suggestion
class OrganizationSamplingProjectRateSerializer(serializers.Serializer):
    id = serializers.IntegerField(required=True)
    sampleRate = serializers.FloatField(required=True, min_value=0, max_value=1)
```

Also we might need to include the project name in the serializer too, since I am not sure if on the frontend we have info about all the projects in the org (i believe we only know about the ones user has access to)

Reply:
The frontend doesn't need the project name (this implements the spec in https://github.com/getsentry/projects/issues/296). 

More importantly, the project slug is considered sensitive information in organizations with closed membership. In that case I'd have to make `get_projects` work with pagination or duplicate all the team filtering logic into the endpoint, which I'd like to avoid.

Reply:
> Why not have just one serializer, and do the logic from get_attrs inside the get method?

That's a great question! I replicated what I see in other endpoints and noticed we always use `sentry.api.serializers.Serializer` to render output rather than the one from `rest_framework`. The logic around batched `get_attrs` seemed very elegant, but I can of course also put this in the endpoint.

Is there general guidance on how to approach this that I may have missed in develop docs?

Reply:
>  The logic around batched get_attrs seemed very elegant
Agree with that, it makes sense to keep it this way

> Is there general guidance on how to approach this that I may have missed in develop docs?
I think that this is a way the docs recommend